### ISSUE TITLE
Number from Argentina who have Signal installed cannot contact each other

### ISSUE THREAD

**Issue by SexyPantz** [signalapp/Signal-Android#10506]
- [x ] I have searched open and closed issues for duplicates
- [x ] I am submitting a bug report for existing functionality that does not work as intended
- [x ] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x ] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I found a pretty troublesome bug involving numbers from Argentina,  specifically number that start with +54 11 and +54 9 11. Most people have the contact written down as 11 xxxx-xxxx . This format works for fine in most other messaging app (Telegram, WhatsApp etc) Those app will usually add the prefix +54 9 which is the prefix needed for inbound international calls. However sometime numbers end up being saved as +54 11 xxxx-xxxx, which is also a valid number for international inbound SMS text, (which is what most apps used to validate the users phone number).
Signal apparently thinks +54 11 xxxx-xxxx and +54 9 11 xxxx-xxxx are two distinct phone numbers, when in practice they are the same. The end result is that depending on how the contact number is saved Signal might not recognise the number as a valid Signal profile and the users wont be able to communicate with each other.

Proposed quick fix:
If the user knows which contact is not showing up they can try changing the number to 11 xxxx xxxx, +54 11 xxxx xxxx or +54 9 11 xxxxx xxxx, refreshing contacts and just figuring out which one works by trial and error.

Long term fix
Update Signal to recognise 11 xxxx xxxx, +54 11 xxxx xxxx and +54 9 11 xxxxx xxxx as the same number. It's just a regex fix, it's not that tricky

### Device info

**Device:** Samsung Galaxy J5 Prime (SM-G570M) 
**Android version:** 8.0.0
**Signal version:** 5.0.8

**Comment by grandchild** [comment]
There's a similar thing [already implemented for Mexico](https://github.com/signalapp/Signal-Android/blob/master/app/src/main/java/org/thoughtcrime/securesms/contacts/sync/FuzzyPhoneNumberHelper.java#L98). So yeah, shouldn't be too hard to add another case for Argentina.

**Comment by mdaloia** [comment]
This is really an adoption drawback (at least on Argentina) on times where people is running out from WhatsApp due to the updated Terms of Service and Privacy policy to be effective on Feb 8th. Telegram (and WhatsApp) have it resolved (or at least never had an issue).

Looking on past issues I found several attempts to address it from quite a long time (2014): #1511 #4378 #5270 #5363 #7668 #10059
Is this something only affecting Android App? On iOS repo found just 2 issues from like 4 and 6 years ago.

It happened to myself. I have all my contacts on my phonebook in an international E.164 form (even the local ones). Registered to Signal using the 9. My friends not, so I can not see them and they can not see me unless update on their phonebook my number to add the 9.

A few people knows that the most correct way to reference a phone in Argentina following E.164 is:
- +54 9 area_code number  (for mobile phones)
- +54 area_code number  (for landline phones)

I'm far from being an expert on the subject but I think that the best way is to normalize the inputs (on a best-effort basis) to the E.164 form and only after it compare to produce a match (or not).

Isn't libphonenumber of any help on this?

**Comment by tandiljuan** [comment]
This is a really old issue in Signal. I guess it isn't being fixing because not all countries have the double international code feature/issue/headache.

Here are two old related (and closed) issues:

- https://github.com/signalapp/Signal-Android/issues/1511
- https://github.com/signalapp/Signal-Android/issues/4378

Here a related issue but for Brazil:

- https://github.com/signalapp/Signal-Android/issues/10854

**Comment by tandiljuan** [comment]
Any idea how can we have the fix implemented ASAP? Thanks!

**Comment by tandiljuan** [comment]
You're right @greyson-signal . In Argentina `+549` is the international code for mobile lines and `+54` is the international code for land lines ([more information](https://en.wikipedia.org/wiki/Telephone_numbers_in_Argentina#Inbound)). So, for registration purpose, `+549` must be preferred.

**Comment by greyson-signal** [comment]
Sounds good, this will be in the next release with some relevant modifications. Thanks!!
